src/auth/test/auth.service.spec.ts

```diff
@@ -201,106 +131,110 @@ describe('AuthService', () => {
       //given
       const userId = faker.number.int({ max: MAX_INT32 });
       const newPassword = faker.internet.password();
-      const spy = jest.spyOn(userRepository, 'updateUserById');
+      jest.spyOn(authClient, 'request').mockResolvedValue(userId);
 
       //when
-      await authService.changePassword(userId, newPassword);
+      const changedPasswordUserId = await authService.changePassword(
+        userId,
+        newPassword,
+      );
 
       //then
-      expect(spy).toHaveBeenCalled();
+      expect(changedPasswordUserId).toEqual(userId);
     });
   });
 
   describe('Reset password', () => {
     it('should generate password reset token', async () => {
       //given
       const email = faker.internet.email();
-
-      jest.spyOn(userRepository, 'getUserByEmail').mockImplementationOnce(() =>
-        Promise.resolve({
-          id: faker.number.int(),
-          email,
-          password: faker.internet.password(),
-          role: Role.USER,
-          activated: false,
-          twoFactorAuth: null,
-          accountActivationToken: null,
-        }),
-      );
+      jest.spyOn(userRepository, 'getUserByEmail').mockResolvedValue({
+        id: faker.number.int(),
+        email,
+        role: Role.USER,
+      });
+      const token = faker.string.alphanumeric({ length: 64 });
+      jest.spyOn(authClient, 'request').mockResolvedValue(token);
 
       //when
-      const token = await authService.generateResetPasswordToken(email);
+      const generatedToken = await authService.generateResetPasswordToken(
+        email,
+      );
 
       //then
-      expect(token).not.toBe('');
-      expect(typeof token).toBe('string');
+      expect(generatedToken).toEqual(token);
     });
   });
 
   describe('Create QR Code', () => {
-    it('should change password', async () => {
+    it('should create QR Code for 2FA', async () => {
       //given
       const userId = faker.number.int();
+      const qrCodeUrl = faker.internet.url();
+      jest.spyOn(authClient, 'request').mockResolvedValue({ qrCodeUrl });
 
       //when
       const qrCode = await authService.createQrCodeFor2fa(userId);
 
       //then
-      expect(typeof qrCode).toBe('string');
+      expect(qrCode).toEqual(qrCodeUrl);
+    });
+  });
+
+  describe('Generate 2FA recovery keys', () => {
+    it('should generate recovery keys for 2FA', async () => {
+      const userId = faker.number.int();
+      const recoveryKeys = Array.from(
+        { length: NUMBER_OF_2FA_RECOVERY_TOKENS },
+        () => faker.string.alphanumeric(),
+      );
+      jest.spyOn(authClient, 'request').mockResolvedValue({ recoveryKeys });
+
+      const generatedRecoveryKeys = await authService.generate2faRecoveryKeys(
+        userId,
+      );
+
+      expect(generatedRecoveryKeys).toEqual(recoveryKeys);
     });
   });
 
   describe('Enable 2fa', () => {
     it('should enable 2fa', async () => {
```

Comment:
You could do `ctrl + f` in IDE and find all instances of `"2fa"` as strings / descriptions etc. and change them to `"2FA"` ("2FA" in strings, "2fa" in object / class / variable names etc.)